unittests/Backend.cpp

```diff
@@ -2214,18 +2214,23 @@ TEST_F(BackendTest, cacheIntegration)
                     ++lgrInfo.seq;
                     lgrInfo.parentHash = lgrInfo.hash;
                     std::srand(std::time(nullptr));
-                    std::shuffle(
-                        lgrInfo.txHash.begin(),
-                        lgrInfo.txHash.end(),
-                        std::default_random_engine(seed));
-                    std::shuffle(
-                        lgrInfo.accountHash.begin(),
-                        lgrInfo.accountHash.end(),
-                        std::default_random_engine(seed));
-                    std::shuffle(
-                        lgrInfo.hash.begin(),
-                        lgrInfo.hash.end(),
-                        std::default_random_engine(seed));
+                    // make sure the hash is unique
+                    auto uniqueHashGenerator =
+                        [seed](ripple::uint256& u256Hash) mutable {
+                            static std::unordered_set<std::string> hashSet;
+                            do
+                            {
+                                std::shuffle(
+                                    u256Hash.begin(),
+                                    u256Hash.end(),
+                                    std::default_random_engine(seed));
```

Comment:
Let's not recreate the generator with the same seed everytime - this yields same sequence everytime. We should create it once and reuse.